tests/test_main.py

```diff
@@ -1,6 +1,260 @@
+        {
+            "name": "Alisa",
+            "age": 19,
+            "wearing_a_mask": True,
+        },
+    ]
+)
+def test_cafe_visit_should_raise_not_vaccinated_error_when_person_does_not_have_vaccine_key(visitor):
+    cafe = Cafe("")
+    with pytest.raises(NotVaccinatedError):
+        cafe.visit_cafe(visitor)
+
+
+@pytest.mark.parametrize(
+    "visitor",
+    [
+        {
+            "name": "John",
+            "age": 21,
+            "vaccine": {
+                "name": "Pfizer",
+                "expiration_date": (datetime.datetime.now() - datetime.timedelta(days=2)).date(),
+            },
+            "wearing_a_mask": True,
+        },
+        {
+            "name": "Alisa",
+            "age": 19,
+            "vaccine": {
+                "name": "Moderna",
+                "expiration_date": (datetime.datetime.now() - datetime.timedelta(days=4)).date(),
+            },
+            "wearing_a_mask": True,
+        },
+    ]
+)
+def test_cafe_visit_should_raise_outdated_vaccine_error_when_vaccine_has_expired(visitor):
+    cafe = Cafe("")
+    with pytest.raises(OutdatedVaccineError):
+        cafe.visit_cafe(visitor)
+
+
+@pytest.mark.parametrize(
+    "visitor",
+    [
+        {
+            "name": "John",
+            "age": 21,
+            "vaccine": {
+                "name": "Pfizer",
+                "expiration_date": (datetime.datetime.now() + datetime.timedelta(days=2)).date(),
+            },
+            "wearing_a_mask": False,
+        },
+        {
+            "name": "Alisa",
+            "age": 19,
+            "vaccine": {
+                "name": "Moderna",
+                "expiration_date": (datetime.datetime.now() + datetime.timedelta(days=4)).date(),
+            },
+            "wearing_a_mask": False,
+        },
+    ]
+)
+def test_cafe_visit_should_raise_not_wearing_mask_error_when_wearing_a_mask_is_false(visitor):
+    cafe = Cafe("")
+    with pytest.raises(NotWearingMaskError):
+        cafe.visit_cafe(visitor)
+
+
+@pytest.mark.parametrize(
+    "visitor,cafe_name",
+    [
+        (
+            {
+                "name": "John",
+                "age": 21,
+                "vaccine": {
+                    "name": "Pfizer",
+                    "expiration_date": datetime.datetime.today().date(),
+                },
+                "wearing_a_mask": True,
+            },
+            "KFC"
+        ),
+        (
+            {
+                "name": "Alisa",
+                "age": 19,
+                "vaccine": {
+                    "name": "Moderna",
+                    "expiration_date": (datetime.datetime.now() + datetime.timedelta(days=4)).date(),
+                },
+                "wearing_a_mask": True,
+            },
+            "McDonald's",
+        )
+    ]
+)
+def test_cafe_visit_should_return_welcome_when_visitor_is_wearing_a_mask_and_vaccinated(visitor, cafe_name):
+    cafe = Cafe(cafe_name)
+    assert cafe.visit_cafe(visitor) == f"Welcome to {cafe_name}"
```

Comment:
Maybe add 'expected_message' to mark.parametrize and assert '... == expected_message'. Not really sure about this.

Reply:
Both ways are ok I believe

Reply:
Thanks, fixed